web/src/types/software.ts

```diff
@@ -129,9 +129,28 @@ type RegisteredAddonInfo = {
   registrationCode: string;
 };
 
+type Conflict = {
+  id: number;
+  description: string;
+  details: string | null;
+  solutions: Solution[];
+};
+
+type Solution = {
+  id: number;
+  description: string;
+  details: string | null;
+};
+
+type SolutionPatch = {
```

Comment:
NP: IMHO, the "Patch" suffix feels too coupled to the underlying HTTP verb used to send the user’s chosen solution to the server. I'd suggest something like ConflictSolution, which better expresses the intent or domain concept.